lib/dpl/provider/pypi.rb

```diff
@@ -38,9 +38,30 @@ def pypi_skip_existing_option
           end
       end
 
+      def pypi_pin_setuptools_version
+        setuptools_version = options[:setuptools_version] || context.env['SETUPTOOLS_VERSION'] || ''
+        if !setuptools_version.empty?
+          '=' + setuptools_version
+        end
+      end
+
+      def pypi_pin_twine_version
+        twine_version = options[:twine_version] || context.env['TWINE_VERSION'] || ''
+        if !twine_version.empty?
+          '=' + twine_version
+        end
+      end
+
+      def pypi_pin_wheel_version
+        wheel_version = options[:wheel_version] || context.env['WHEEL_VERSION'] || ''
+        if !wheel_version.empty?
+          '=' + wheel_version
+        end
+      end
+
       def install_deploy_dependencies
         unless context.shell "wget -O - https://bootstrap.pypa.io/get-pip.py | python - --no-setuptools --no-wheel && " \
-                             "pip install --upgrade setuptools twine wheel"
+                             "pip install --upgrade setuptools#{pypi_pin_setuptools_version} twine#{pypi_pin_twine_version} wheel#{pypi_pin_wheel_version}"
```

Comment:
Tacking on `*_version` return values to the `pip install` arguments is a bit strange. Instead, name the methods differently; e.g., `pypi_setuptools_arg`, which returns `setuptools` by default, but `setuptools=X.Y.Z` when option is given.

Reply:
Good idea - will do that.